spyder/plugins/switcher/plugin.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2009- Spyder Project Contributors
+#
+# Distributed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Switcher Plugin.
+"""
+
+# Third-party imports
+
+# Local imports
+from spyder.api.translations import _
+from spyder.api.plugins import Plugins, SpyderPluginV2
+from spyder.api.plugin_registration.decorators import on_plugin_available, on_plugin_teardown
+from spyder.plugins.switcher.container import SwitcherContainer
+from spyder.plugins.mainmenu.api import ApplicationMenus, FileMenuSections
+
+
+# --- Constants
+# ----------------------------------------------------------------------------
+class SwitcherActions:
+    FileSwitcherAction = 'file switcher'
+    SymbolFinderAction = 'symbol finder'
+
+
+# --- Plugin
+# ----------------------------------------------------------------------------
+class Switcher(SpyderPluginV2):
+    """
+    Switcher plugin.
+    """
+
+    NAME = "switcher"
+    OPTIONAL = [Plugins.MainMenu]
+    CONTAINER_CLASS = SwitcherContainer
+    CONF_SECTION = NAME
+    CONF_FILE = False
+
+    # --- Signals
+
+    # --- SpyderPluginV2 API
+    # ------------------------------------------------------------------------
+    @staticmethod
+    def get_name():
+        return _("Switcher")
+
+    def get_description(self):
+        return _("A multi-purpose switcher.")
+
+    def get_icon(self):
+        return self.create_icon('filelist')
+
+    def on_initialize(self):
+        container = self.get_container()
+    
+    @on_plugin_available(plugin=Plugins.MainMenu)
```

Comment:
To remove some whitespace in a blank line:
```suggestion
        container = self.get_container()

    @on_plugin_available(plugin=Plugins.MainMenu)
```